src/plugins/ramps/paybis/paybisRampPlugin.ts

```diff
@@ -0,0 +1,1360 @@
+        const { name, pairs } = paymentMethodPairs
+        if (name == null) continue
+        const edgePaymentType = PAYMENT_METHOD_MAP[name]
+        if (edgePaymentType == null) continue
+        for (const pair of pairs) {
+          const { fromAssetId, to } = pair
+
+          let paymentMethodObj = allowedCurrencyCodes.sell[edgePaymentType]
+          if (paymentMethodObj == null) {
+            paymentMethodObj = { providerId: pluginId, crypto: {}, fiat: {} }
+            allowedCurrencyCodes.sell[edgePaymentType] = paymentMethodObj
+          }
+
+          const edgeTokenId = PAYBIS_TO_EDGE_CURRENCY_MAP[fromAssetId]
+          if (edgeTokenId == null) continue
+          const { pluginId: currencyPluginId } = edgeTokenId
+
+          let tokens = paymentMethodObj.crypto[currencyPluginId]
+          if (tokens == null) {
+            tokens = []
+            paymentMethodObj.crypto[currencyPluginId] = tokens
+          }
+          addTokenToArray({ tokenId: edgeTokenId.tokenId }, tokens)
+
+          for (const fiat of to) {
+            paymentMethodObj.fiat[`iso:${fiat}`] = true
+          }
+        }
+      }
+    }
+  }
+
+  const ensureStateInitialized = async (): Promise<void> => {
+    if (state == null) {
+      const { apiKey, partnerUrl, privateKeyB64 } = initOptions
+
+      let partnerUserId: string
+      if (pluginConfig.store != null) {
+        partnerUserId = await pluginConfig.store
+          .getItem('partnerUserId')
+          .catch(() => '')
+        if (partnerUserId === '' && pluginConfig.makeUuid != null) {
+          partnerUserId = await pluginConfig.makeUuid()
+          await pluginConfig.store.setItem('partnerUserId', partnerUserId)
+        } else if (partnerUserId === '') {
+          partnerUserId = `edge-user-${Date.now()}-${Math.random()
+            .toString(36)
+            .substring(7)}`
+          await pluginConfig.store.setItem('partnerUserId', partnerUserId)
+        }
+      } else {
+        partnerUserId = `edge-user-${Date.now()}-${Math.random()
+          .toString(36)
+          .substring(7)}`
+      }
+
+      state = {
+        apiKey,
+        partnerUrl,
+        privateKeyB64,
+        partnerUserId
+      }
+    }
+  }
+
+  const ensureAssetsInitialized = async (
+    direction: 'buy' | 'sell'
+  ): Promise<void> => {
+    await ensureStateInitialized()
+
+    if (direction === 'buy') {
+      await initializeBuyPairs()
+    } else {
+      await initializeSellPairs()
+    }
+  }
+
+  const validateSupportRequest = (
+    regionCode?: FiatPluginRegionCode,
+    countryCode?: string
+  ): { supported: false } | undefined => {
+    // Check region restrictions
+    if (regionCode != null) {
+      try {
+        validateRegion(pluginId, regionCode, SUPPORTED_REGIONS)
+      } catch (error) {
+        return { supported: false }
+      }
+    }
+
+    // Check country-specific restrictions
+    if (countryCode === 'GB') {
```

Comment:
Just put this restriction in `SUPPORTED_REGIONS`. Then you can just call validateRegion directly and no need for `validateSupportRequest`

Reply:
The condition `countryCode === 'GB'` is already handled by `validateRegion` so it can be removed. This makes `validateSupportRequest` just a wrapper around `validateRegion` to catch the errors thrown and return `undefined` for valid region support and `{ supported: false }` for invalid region. This is a confusing signature. The wrapper isn't really necessary, we can just inline `validateRegion` in the two places where `validateSupportRequest` is invoked in the paybis plugin and handle the error catching accordingly.